src/lib/device-types/index.ts

```diff
@@ -246,21 +329,34 @@ export const validateSlug = (slug?: string) => {
 	return slug;
 };
 
-export const deviceTypes = (): Promise<DeviceType[]> => {
-	return getDeviceTypes().then(deviceTypesInfos => {
-		// exclude aliases
-		return _(deviceTypesInfos)
-			.filter(
-				(deviceTypesInfo, slug) =>
-					deviceTypesInfo.latest.deviceType.slug === slug,
-			)
-			.map(deviceTypesInfo => deviceTypesInfo.latest.deviceType)
-			.value();
-	});
+export const deviceTypes = (api: PinejsClient): Promise<DeviceType[]> => {
+	return Promise.join(
+		getDeviceTypes(),
+		getAccessibleSlugs(api),
+		(
+			deviceTypesInfos: Dictionary<DeviceTypeInfo>,
+			accessibleDeviceTypes: string[],
+		) => {
+			const deviceTypes: DeviceType[] = _(deviceTypesInfos)
+				.filter(
+					(deviceTypesInfo: DeviceTypeInfo, slug: string) =>
+						deviceTypesInfo.latest.deviceType.slug === slug,
+				)
+				.map(deviceTypesInfo => deviceTypesInfo.latest.deviceType)
+				.value();
+			const accessSet = new Set(accessibleDeviceTypes);
+			return _.filter(deviceTypes, (o: DeviceType) => {
+				return accessSet.has(o.slug);
```

Comment:
How about moving this in the lodash chained filter above, instead of having a separate `_.filter()` call?
```ts
const accessSet = new Set(accessibleDeviceTypes);
const deviceTypes: DeviceType[] = _(deviceTypesInfos)
	.filter(
		(deviceTypesInfo: DeviceTypeInfo, slug: string) => {
			const dtSlug = deviceTypesInfo.latest.deviceType.slug;
			return dtSlug === slug && accessSet.has(dtSlug),
		}
	)
	.map(deviceTypesInfo => deviceTypesInfo.latest.deviceType)
	.value();
```
PS: I was probably not clear enough earlier.